### DISCUSSION TITLE
Invert mask toggle for inpaint?

### DISCUSSION THREAD

**Discussion by aolko** [lllyasviel/Fooocus#2482] [Ideas]
Add an invert toggle for inpaint, so we don't have to do the tedious work of masking everything __but__ the object when we need to change background

**Comment by ILikeToasters** [comment]
In the Developer Debug Mode in the Inpaint tab at the bottom. <img width="227" alt="Screenshot 2024-03-08 171428" src="https://github.com/lllyasviel/Fooocus/assets/4357657/3c5090a1-df88-4d96-8884-1652a8e13095">

**Reply by aolko** [reply]
why not just put it in the inpaint tab?

**Reply by mashb1t** [reply]
Not everybody needs it, so this can be optionally activated in developer debug mode.

**Reply by ED-robert** [reply]
I'm using 2.5.5 which apparently was the last version, and I do not see that option?

![image](https://github.com/user-attachments/assets/d0e12a01-fdc6-4cfe-99a9-44d43cd361e9)

Anyone know what I'm missing?

Also, the issue for this is still open:

https://github.com/lllyasviel/Fooocus/issues/1404

Further, some sites with tutorials suggest that the painted area is the one getting re-generated, but like the first post says here, that's not what I see, rather the painted area is what gets un-altered which is very unintuitive and tedious:

https://www.promptzone.com/muhsin/mastering-fooocus-inpainting-revolutionize-your-image-editing-47dd

UPDATE: Re-downloading the latest release and just unpacking it on top of what was already installed now suddenly has inverted the behavior... but the button for it still has not appeared.

**Reply by mashb1t** [reply]
@aolko after receiving feedback i've moved that option directly to inpainting

![Screenshot 2025-04-14 at 15 07 47](https://github.com/user-attachments/assets/eade8d0a-085a-4916-a2ea-6facf6e5ec45)

**Reply by ED-robert** [reply]
Thank you, I see it now.
